### PR TITLE
[codenav]: refactor of upload resolver 

### PR DESCRIPTION
**Pull request by Numbers88s**
Closes https://github.com/sourcegraph/sourcegraph/issues/38160
Refactor of upload resolver as part of our ongoing efforts towards RFC:619

## Test plan
yeah

### CODE REVIEW

File: enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/prefetcher.go [hunk 1]
```diff
@@ -71,10 +71,14 @@ func (p *Prefetcher) GetUploadByID(ctx context.Context, id int) (store.Upload, b
 	}
 	sort.Ints(ids)
 
-	uploads, err := p.resolver.GetUploadsByIDs(ctx, ids...)
+	uploadsResolver := p.resolver.UploadsResolver()
+
+	u, err := uploadsResolver.GetUploadsByIDs(ctx, ids...)
```
Comment (Strum355) [inline enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/prefetcher.go:76]:
```suggestion
	u, err := p.resolver.UploadsResolver().GetUploadsByIDs(ctx, ids...)
```

File: enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/resolver.go [hunk 2]
```diff
@@ -159,8 +161,8 @@ func (r *Resolver) DeleteLSIFUpload(ctx context.Context, args *struct{ ID graphq
 	if err != nil {
 		return nil, err
 	}
-
-	if err := r.resolver.DeleteUploadByID(ctx, int(uploadID)); err != nil {
+	uploadResolver := r.resolver.UploadsResolver()
+	if _, err := uploadResolver.DeleteUploadByID(ctx, int(uploadID)); err != nil {
```
Comment (Strum355) [inline enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/resolver.go:165]:
```suggestion
	if _, err := r.resolver.UploadsResolver().DeleteUploadByID(ctx, int(uploadID)); err != nil {
```

File: enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/resolver.go [hunk 3]
```diff
@@ -276,7 +278,10 @@ func (r *Resolver) CommitGraph(ctx context.Context, id graphql.ID) (_ gql.CodeIn
 		return nil, err
 	}
 
-	return r.resolver.CommitGraph(ctx, int(repositoryID))
+	uploadResolver := r.resolver.UploadsResolver()
+	commitGraphResolver := uploadResolver.CommitGraphResolverFromFactory(ctx, int(repositoryID))
```
Comment (Strum355) [inline enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/resolver.go:282]:
```suggestion
	commitGraphResolver := r.resolver.UploadsResolver().CommitGraphResolverFromFactory(ctx, int(repositoryID))
```

File: enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/resolver.go [hunk 4]
```diff
@@ -140,8 +140,10 @@ func (r *Resolver) LSIFUploadsByRepo(ctx context.Context, args *gql.LSIFReposito
 	// Create a new prefetcher here as we only want to cache upload and index records in
 	// the same graphQL request, not across different request.
 	prefetcher := NewPrefetcher(r.resolver)
+	uploadResolver := r.resolver.UploadsResolver()
+	uploadConnectionResolver := uploadResolver.UploadsConnectionResolverFromFactory(opts)
```
Comment (Strum355) [inline enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/resolver.go:144]:
```suggestion
	uploadConnectionResolver := r.resolver.UploadsResolver().UploadsConnectionResolverFromFactory(opts)
```

File: enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/upload.go [hunk 5]
```diff
@@ -149,7 +149,8 @@ func (r *UploadResolver) DocumentPaths(ctx context.Context, args *gql.LSIFUpload
 	if args.Pattern != "" {
 		pattern = args.Pattern
 	}
-	documents, totalCount, err := r.resolver.GetUploadDocumentsForPath(ctx, r.upload.ID, pattern)
+	uploadsResolver := r.resolver.UploadsResolver()
+	documents, totalCount, err := uploadsResolver.GetUploadDocumentsForPath(ctx, r.upload.ID, pattern)
```
Comment (Strum355) [inline enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/upload.go:153]:
```suggestion
	documents, totalCount, err := r.resolver.UploadsResolver().GetUploadDocumentsForPath(ctx, r.upload.ID, pattern)
```

File: enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/upload.go [hunk 6]
```diff
@@ -161,7 +162,8 @@ func (r *UploadResolver) DocumentPaths(ctx context.Context, args *gql.LSIFUpload
 }
 
 func (r *UploadResolver) AuditLogs(ctx context.Context) (*[]gql.LSIFUploadsAuditLogsResolver, error) {
-	logs, err := r.resolver.AuditLogsForUpload(ctx, r.upload.ID)
+	uploadResolver := r.resolver.UploadsResolver()
+	logs, err := uploadResolver.GetAuditLogsForUpload(ctx, r.upload.ID)
```
Comment (Strum355) [inline enterprise/cmd/frontend/internal/codeintel/resolvers/graphql/upload.go:166]:
```suggestion
	logs, err := r.resolver.UploadsResolver().GetAuditLogsForUpload(ctx, r.upload.ID)
```

File: internal/codeintel/stores/dbstore/dependency_index.go [hunk 7]
```diff
@@ -58,14 +58,6 @@ var dependencySyncingJobColumns = []*sqlf.Query{
...
 }
 
-// scanDependencySyncingJobs scans a slice of dependency syncing jobs from the return value of
-// `*Store.query`.
-var scanDependencySyncingJobs = basestore.NewSliceScanner(scanDependencySyncingJob)
-
-// scanFirstDependencySyncingingJob scans a slice of dependency indexing jobs from the return
-// value of `*Store.query` and returns the first.
-var scanFirstDependencySyncingingJob = basestore.NewFirstScanner(scanDependencySyncingJob)
-
```
Comment (Strum355) [inline internal/codeintel/stores/dbstore/dependency_index.go:68]: Where have these and the ones below been moved to? Not showing up in a search

**Review by Strum355 — APPROVED** [review]
Think that all makes sense, some things im not sure where theyve been moved to

File: internal/codeintel/stores/dbstore/dependency_index.go (hunk 7 above)
Comment (efritz) [reply internal/codeintel/stores/dbstore/dependency_index.go:68]: They might not be read directly

File: internal/codeintel/uploads/internal/lsifstore/testdata/lsif-go@ad3507cb.sql [hunk 8]
```diff
@@ -0,0 +1,2319 @@
+--
```
Comment (efritz) [inline internal/codeintel/uploads/internal/lsifstore/testdata/lsif-go@ad3507cb.sql:1]: Where did this come from?

**Review by efritz — APPROVED** [review]

File: internal/codeintel/uploads/internal/lsifstore/testdata/lsif-go@ad3507cb.sql (hunk 8 above)
Comment (Numbers88s) [reply internal/codeintel/uploads/internal/lsifstore/testdata/lsif-go@ad3507cb.sql:1]: its part of some of the lsifstore test

**Pull request merged**